tests/tt_eager/ops/test_sliding_window_ops.cpp

```diff
@@ -0,0 +1,489 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include <sys/wait.h>
+#include "common/bfloat16.hpp"
+#include "tensor/host_buffer/functions.hpp"
+#include "tensor/host_buffer/types.hpp"
+#include "tensor/types.hpp"
+#include "tt_metal/host_api.hpp"
+#include "tt_eager/tensor/tensor.hpp"
+#include "tt_numpy/functions.hpp"
+#include "tt_dnn/op_library//sliding_window_op_infra/sliding_window.hpp"
+
+#include "tensor/tensor.hpp"
+
+#include <cstdint>
+#include <cstdlib>
+#include <numeric>
+#include <tuple>
+
+using tt::tt_metal::Tensor;
+using tt::tt_metal::Shape;
+
+
+vector<float> create_filter_vector(owned_buffer::Buffer<bfloat16> &filter_tensor_buf, uint32_t filter_h, uint32_t filter_w) {
+    vector<float> filter_vector;
+    for(auto h = 0; h < filter_h; h++) {
+        for(auto w = 0; w < filter_w; w++) {
+            filter_vector.push_back(filter_tensor_buf[ h * filter_w + w].to_float());
+        }
+    }
+    return filter_vector;
+}
+
+owned_buffer::Buffer<bfloat16> ref_conv_op(
```

Comment:
It would be good to keep all the reference calculation functions in a separate cpp file under the `sliding_window_op_infra` directory instead of in this test.